src/components/icon/Icon.tsx

```diff
@@ -29,13 +29,15 @@ const Icon = ({ children, className, color = 'teal', tint = 'normal', opacity =
       {React.Children.map(children, (child) => {
         // Check if child is an actual React component
         // if so, pass the needed props. If not, just render it.
-        if (!child?.type) {
-          return child;
-        }
+        if (React.isValidElement(child)) {
+          if (!child?.type) {
+            return child;
+          }
 
-        return React.cloneElement(child, {
-          opacity,
-        });
+          return React.cloneElement(child, {
+            opacity,
+          });
+        }
       })}
```

Comment:
I think in order to keep the old behaviour you need to return the child if it's not a valid element.
And you can probably drop the `child.type` bit since there should always be a type if it's a valid element

Reply:
Yeah, thanks for the remark. Dropped the child type and checked first if its not a valid element 👍